infra/MyStack.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Pulumi;
+using Pulumi.AzureNative.Resources;
+using Pulumi.AzureNative.Web;
+using Pulumi.AzureNative.Web.Inputs;
+
+// Define global variables
+
+class MyStack : Stack
+{
+    const string repoUrl = "https://github.com/Mentoring-Bordeaux/nebulift";
+    const string sitePath = "src/front"; // TODO : change the sitePath
+    const string apiPath = "src/back"; // TODO : change the apiPath
+
+    public MyStack()
+    {
+        // Retrieve the current stack name
+        var stackName = Pulumi.Deployment.Instance.StackName;
+
+        // Create an Azure Resource Group
+        var resourceGroup = new ResourceGroup("rg-nebulift-dev-westeurope-");
+
+        // Create an App Service Plan
+        var appServicePlan = new AppServicePlan("wapp-plan-nebulift-dev-westeurope-", new AppServicePlanArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            Location = resourceGroup.Location,
+            Sku = new SkuDescriptionArgs
+            {
+                Name = stackName == "prod" ? "B1" : "F1",
+                Tier = stackName == "prod" ? "Basic" : "Free"
+            }
+        });
+
+        // Create an App Service
+        var appService = new WebApp("wapp-nebulift-dev-westeurope-", new WebAppArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            Location = resourceGroup.Location,
+            ServerFarmId = appServicePlan.Id
+        });
+
+        // Export the App Service URL
+        this.Endpoint = Output.Format($"https://{appService.DefaultHostName}");
+
+        // Create a Static Web App
+        var staticWebApp = new StaticSite("stapp-nebulift-dev-westeurope-", new StaticSiteArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            Location = resourceGroup.Location,
+            RepositoryUrl = repoUrl,
+            Branch = "main",
+            BuildProperties = new StaticSiteBuildPropertiesArgs
+            {
+                AppLocation = sitePath,
+                ApiLocation = apiPath,
+                OutputLocation = "build"
+            },
+            Sku = new SkuDescriptionArgs
+            {
+                Name = "Free"
+            }
+        });
+
```

Comment:
You probably will need to link the API to the static web app to make easy communication between them: https://www.pulumi.com/registry/packages/azure-native/api-docs/web/staticsitelinkedbackend/